src/Raven.Server/Documents/Indexes/MapReduce/ReduceMapResultsBase.cs

```diff
@@ -426,9 +426,13 @@ private void WriteLastEtags(TransactionOperationContext indexContext)
 
                         using (var result = AggregateBranchPage(page, table, indexContext, branchesToAggregate, compressedEmptyLeafs, failedAggregatedLeafs, tree, token))
                         {
-                            if (parentPage == -1)
+                            if (parentPage == -1 )
                             {
-                                IndexReducedResult(indexContext, stats, writer, reduceKeyHash, result.GetOutputs(), false);
+                                if (branchesToAggregate.Count == 0 || // no more branches
+                                    branchesToAggregate.Count == 1 && branchesToAggregate.Contains(pageNumber)) // just this one, which we'll shortly remove
```

Comment:
Here we have the change that isn't specific only to Corax but affects the reduce process in general.

The reason for that is that Corax doesn't really like if we put same entry multiple times so here we're avoiding doing so. With current impl we're supposed to skip the writes to an index that would be overwritten by further aggregations of branches.

Reply:
For Lucene, that is a valid change as well, since it reduce the amount of work that is required.